sms_sponsorship/models/sms_child_request.py

```diff
@@ -122,19 +122,24 @@ def _compute_full_url(self):
     @api.multi
     @api.depends("date")
     def _compute_event(self):
-        limit_date = datetime.today() - relativedelta(days=7)
-        for request in self.filtered(lambda r: r.source == "SMS" and r.date):
-            event_id = self.env["crm.event.compassion"].search(
-                [
-                    ("accepts_sms_booking", "=", True),
-                    ("start_date", "<=", request.date),
-                    ("start_date", ">=", limit_date),
-                ],
-                order="start_date desc",
-                limit=1,
-            )
-            # event_id is None if start_date of most recent event is>1 week old
-            request.event_id = event_id
+        for request in self:
+            if request.child_id.hold_event:
+                request.event_id = request.child_id.hold_event
+            else:
+                limit_date = datetime.today() - relativedelta(days=7)
+                for sms_request in self.filtered(lambda r: r.source == "SMS" and r.date):
+                    event_id = self.env["crm.event.compassion"].search(
+                        [
+                            ("accepts_sms_booking", "=", True),
+                            ("start_date", "<=", sms_request.date),
+                            ("start_date", ">=", limit_date),
+                        ],
+                        order="start_date desc",
+                        limit=1,
+                    )
+                    # event_id is None if start_date of most recent event is > 1 week old
+                    sms_request.event_id = event_id
```

Comment:
No need to loop twice on the records. Last line is for v14 compatibility.
```suggestion
            elif request.source == "SMS" and request.date:
                limit_date = datetime.today() - relativedelta(days=7)
                event_id = self.env["crm.event.compassion"].search(
                    [
                        ("accepts_sms_booking", "=", True),
                        ("start_date", "<=", request.date),
                        ("start_date", ">=", limit_date),
                    ],
                    order="start_date desc",
                    limit=1,
                )
                # event_id is None if start_date of most recent event is > 1 week old
                request.event_id = event_id
            else:
                request.event_id = False
```